### PR TITLE
DOCSP-43067: Insert documents

### PR DESCRIPTION
**Pull request by norareidy**
# Pull Request Info

[PR Reviewing Guidelines](https://github.com/mongodb/docs-golang/blob/master/REVIEWING.md)

JIRA - https://jira.mongodb.org/browse/DOCSP-43067
Staging - https://deploy-preview-44--docs-c.netlify.app/write/insert/

## Self-Review Checklist

- [ ] Is this free of any warnings or errors in the RST?
- [x] Did you run a spell-check?
- [x] Did you run a grammar-check?
- [ ] Are all the links working?
- [ ] Are the [facets and meta keywords](https://wiki.corp.mongodb.com/display/DE/Docs+Taxonomy) accurate?

### CODE REVIEW

File: source/write/insert.txt [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+   :keywords: code example, write, save, create
+
+Overview
+--------
+
+In this guide, you can learn how to use the {+driver-short+} to add
+documents to a MongoDB collection by performing **insert operations**.
+
+An insert operation inserts one or more documents into a MongoDB collection.
+You can perform an insert operation by using the following methods:
```
Comment (mcmorisi) [inline source/write/insert.txt:27]: C doesn't have methods, only functions. Applies to all occurrences on this page.

```suggestion
You can perform an insert operation by using the following methods:
```

File: source/includes/write/insert.c [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    mongoc_init();
+
+    client =
+        mongoc_client_new ("<connection string URI>");
+    collection = mongoc_client_get_collection (client, "sample_restaurants", "restaurants");
+    
+    {
+        // Inserts a document that stores a "name" value of "Mongo's Burgers" into the collection
+        // start-insert-one
+        bson_t *document = BCON_NEW("name", BCON_UTF8("Mongo's Burgers"));
```
Comment (mcmorisi) [inline source/includes/write/insert.c:20]: C documentation puts a space between the function name and parentheses, so we should do the same.

Applies to this whole file.

```suggestion
        bson_t *document = BCON_NEW ("name", BCON_UTF8("Mongo's Burgers"));
```

File: source/write/insert.txt [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+
+.. TODO:
+  For runnable code examples of inserting documents with the {+driver-short+}, see
+  :ref:`php-write`.
+
+API Documentation
+~~~~~~~~~~~~~~~~~
+
+To learn more about any of the methods or types discussed in this
+guide, see the following API documentation:
```
Comment (mcmorisi) [inline source/write/insert.txt:173]:
```suggestion

To learn more about any of the functions discussed in this
guide, see the following API documentation:
```

**Review by mcmorisi — APPROVED** [review]
LGTM with a few C-specific callouts!

**Review by joshbsiegel — CHANGES_REQUESTED** [review]

File: source/includes/write/insert.c [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+        }
+        bson_destroy (document);
+        // end-insert-one
+    }
+
+    {
+        // Inserts documents representing restaurants into the collection
+        // start-insert-many
+        size_t num_docs = 2;
+	    bson_t *docs[num_docs];
```
Comment (joshbsiegel) [inline source/includes/write/insert.c]:
```suggestion
	bson_t *docs[num_docs];
```

**Review by joshbsiegel — APPROVED** [review]
LGTM

**Pull request merged**